### PR TITLE
resource_control: add background task resource control

### PR DESCRIPTION
**Pull request by glorv**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-alter-resource-group.md [hunk 1]
```diff
@@ -59,6 +62,15 @@ ResourceGroupRunawayActionOption ::=
...
 
+BackgroundOptionList ::= 
+    DirectBackgroundOption
+|   BackgroundOptionList DirectBackgroundOption
+|   BackgroundOptionList ',' DirectBackgroundOption
+
+DirectBackgroundOption ::= 
+    "TASK_TYPES" EqOpt stringLit
+
+
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-resource-group.md:73]:
````suggestion

BackgroundOptionList ::= 
    DirectBackgroundOption
|   BackgroundOptionList DirectBackgroundOption
|   BackgroundOptionList ',' DirectBackgroundOption

DirectBackgroundOption ::= 
    "TASK_TYPES" EqOpt stringLit
```
````

File: sql-statements/sql-statement-alter-resource-group.md [hunk 2]
```diff
@@ -67,11 +79,13 @@ TiDB 支持以下 `DirectResourceGroupOption`, 其中 [Request Unit (RU)](/tidb-
 | `PRIORITY`    | 任务在 TiKV 上处理的绝对优先级  | `PRIORITY = HIGH` 表示优先级高。若未指定则默认为 `MEDIUM`。 |
 | `BURSTABLE`   | 允许对应的资源组超出配额后使用空余的系统资源。 |
 | `QUERY_LIMIT` | 当查询执行满足该条件时，识别该查询为 Runaway Query 并执行相应的操作 | `QUERY_LIMIT=(EXEC_ELAPSED='60s', ACTION=KILL, WATCH=EXACT DURATION='10m')` 表示当执行时间超过 60 秒后识别为 Runaway Query，对该查询执行终止操作，并在 10 分钟内对同样的 SQL 直接执行终止操作。`QUERY_LIMIT=()` 或 `QUERY_LIMIT=NULL` 则表示不进行 Runaway 控制。具体参数介绍参见[管理资源消耗超出预期的查询 (Runaway Queries)](/tidb-resource-control.md#管理资源消耗超出预期的查询-runaway-queries)。 ｜
+| `BACKGROUND`  | 后台任务相关的设置。具体参数介绍参见[管理后台任务](/tidb-resource-control.md#管理后台任务) | `BACKGROUND=(TASK_TYPES="br,analyze")` 表示将备份恢复和收集统计信息相关的任务作为后台任务调度 |
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-resource-group.md:82]:
```suggestion
| `BACKGROUND`  | 后台任务相关的设置。具体参数介绍参见[管理后台任务](/tidb-resource-control.md#管理后台任务) | `BACKGROUND=(TASK_TYPES="br,analyze")` 表示将备份恢复和收集统计信息相关的任务作为后台任务调度。 |
```

File: sql-statements/sql-statement-alter-resource-group.md [hunk 3]
```diff
@@ -67,11 +79,13 @@ TiDB 支持以下 `DirectResourceGroupOption`, 其中 [Request Unit (RU)](/tidb-
...
 
 > **注意：**
 > 
 > `ALTER RESOURCE GROUP` 语句只能在全局变量 [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-从-v660-版本开始引入) 参数设置为 `ON` 时才能执行。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-resource-group.md:86]:
```suggestion
> **注意：**
>
> - `ALTER RESOURCE GROUP` 语句只能在全局变量 [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-从-v660-版本开始引入) 参数设置为 `ON` 时才能执行。
```

File: sql-statements/sql-statement-alter-resource-group.md [hunk 4]
```diff
@@ -67,11 +79,13 @@ TiDB 支持以下 `DirectResourceGroupOption`, 其中 [Request Unit (RU)](/tidb-
...
-> `ALTER RESOURCE GROUP` 语句支持以增量方式修改，未指定的参数保持不变。但其中 `QUERY_LIMIT` 作为一个整体，无法部分修改其中的参数。
+> `ALTER RESOURCE GROUP` 语句支持以增量方式修改，未指定的参数保持不变。但其中 `QUERY_LIMIT` 和 `BACKGROUND` 各自作为一个整体，无法部分修改其中的参数。
+> 目前仅 `default` 资源组支持修改 `BACKGROUND` 相关设置。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-resource-group.md:88]:
```suggestion
> - `ALTER RESOURCE GROUP` 语句支持以增量方式修改，未指定的参数保持不变。但其中 `QUERY_LIMIT` 和 `BACKGROUND` 各自作为一个整体，无法部分修改其中的参数。
> - 目前仅 `default` 资源组支持修改 `BACKGROUND` 相关设置。
```

File: sql-statements/sql-statement-alter-resource-group.md [hunk 5]
```diff
@@ -124,11 +138,32 @@ SELECT * FROM information_schema.resource_groups WHERE NAME ='rg1';
...
-| rg1  | 200        | LOW      | YES       | EXEC_ELAPSED=1s, ACTION=COOLDOWN, WATCH=EXACT[30s] |
-+------+------------+----------+-----------+----------------------------------------------------+
++------+------------+----------+-----------+----------------------------------------------------------------+------------+
+| NAME | RU_PER_SEC | PRIORITY | BURSTABLE | QUERY_LIMIT                                                    | BACKGROUND |
++------+------------+----------+-----------+----------------------------------------------------------------+------------+
+| rg1  | 200        | LOW      | NO        | EXEC_ELAPSED='1s', ACTION=COOLDOWN, WATCH=EXACT DURATION='30s' | NULL       |
++------+------------+----------+-----------+----------------------------------------------------------------+------------+
+1 rows in set (1.30 sec)
+```
+
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-resource-group.md:148]:
```suggestion

修改 `default` 资源组的后台任务 `BACKGROUND` 相关设置。

```

File: sql-statements/sql-statement-create-resource-group.md [hunk 6]
```diff
@@ -78,6 +89,7 @@ TiDB 支持以下 `DirectResourceGroupOption`, 其中 [Request Unit (RU)](/tidb-
 >
 > - `CREATE RESOURCE GROUP` 语句只能在全局变量 [`tidb_enable_resource_control`](/system-variables.md#tidb_enable_resource_control-从-v660-版本开始引入) 设置为 `ON` 时才能执行。
 > - TiDB 集群在初始化时会自动创建 `default` 资源组，其 `RU_PER_SEC` 的默认值为 `UNLIMITED` (等同于 `INT` 类型最大值，即 `2147483647`)，且为 `BURSTABLE` 模式。所有未绑定资源组的请求都将自动绑定至此资源组。在新建配置其他资源组时，建议根据实际情况修改 `default` 资源组的配置。
+> 目前仅 `default` 资源组支持修改 `BACKGROUND` 相关设置。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-create-resource-group.md:92]:
```suggestion
> - 目前仅 `default` 资源组支持修改 `BACKGROUND` 相关设置。
```

File: tidb-resource-control.md [hunk 7]
```diff
@@ -333,6 +333,53 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+
+    ```sql
+    ALTER RESOURCE GROUP rg1 BACKGROUND=NULL;
+    ```
+
+4. 修改 `rg1` 资源组，将后台任务的类型设置为空，此时此资源组的所有任务类型都不会作为后台任务处理。
+
+    ```sql
+    ALTER RESOURCE GROUP rg1 BACKGROUND=(TASK_TYPES="");
+    ```
```
Comment (Oreoxmt) [inline tidb-resource-control.md:381]:
````suggestion
`TASK_TYPES`：设置需要作为后台任务管理的任务类型，多个任务类型以 `,` 分隔。

目前 TiDB 支持如下几种后台任务的类型：

- `lightning`：使用 [TiDB Lightning](tidb-lightning/tidb-lightning-overview.md) 执行导入任务。同时支持 TiDB Lightning 的物理和逻辑导入模式。
- `br`：使用 [BR](/br/backup-and-restore-overview.md) 执行数据备份和恢复。目前不支持 PITR。
- `ddl`：对于 Reorg DDL，控制批量数据回写阶段的资源使用。
- `analyze`：对应手动执行或系统自动触发的[收集统计信息](/statistics.md#统计信息的收集)任务。

#### 示例

1. 创建 `rg1` 资源组，并将后台任务类型设置为 `br` 和 `analyze`。

    ```sql
    CREATE RESOURCE GROUP IF NOT EXISTS rg1 RU_PER_SEC = 500 BACKGROUND=(TASK_TYPES='br,analyze');
    ```

2. 修改 `rg1` 资源组，将后台任务类型修改为 `br` 和 `ddl`。

    ```sql
    ALTER RESOURCE GROUP rg1 BACKGROUND=(TASK_TYPES='br,ddl');
    ```

3. 修改 `rg1` 资源组，将后台任务的类型还原为默认值。此时后台任务的类型将使用 `default` 资源组的配置。

    ```sql
    ALTER RESOURCE GROUP rg1 BACKGROUND=NULL;
    ```

4. 修改 `rg1` 资源组，将后台任务的类型设置为空，此时此资源组的所有任务类型都不会作为后台任务处理。

    ```sql
    ALTER RESOURCE GROUP rg1 BACKGROUND=(TASK_TYPES="");
    ```
````

File: tidb-resource-control.md [hunk 8]
```diff
@@ -333,6 +333,53 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
 
+### 管理后台任务
+
+> **警告：**
+>
+> 该功能目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+后台任务是指那些优先级不高但是又需要消耗大量资源的任务，如数据备份和自动统计信息收集等，这些任务通常定期或不定期触发，在执行的时候会消耗大量资源从而对在线的高优先级的任务的性能产生大量影响。
+
+自 v7.4.0 版本开始，TiDB 资源管控引入了对后台任务的管理，当一种任务被标记为 `后台任务`时，TiKV 会对这种任务的执行施加动态的使用资源的限制，以尽量避免此类任务在执行的时候对其他的前台任务的性能产生影响。TiKV 通过实时地监测所有前台任务所消耗的 CPU 和 IO 等资源，并根据实例总的资源上限计算出后台任务可使用的资源阈值，所有后台任务在执行时会受此阈值的限制。
```
Comment (Connor1996) [inline tidb-resource-control.md:344]:
```suggestion
自 v7.4.0 版本开始，TiDB 资源管控引入了对后台任务的管理，当一种任务被标记为 `后台任务` 时，TiKV 会对这种任务的执行施加动态的使用资源的限制，以尽量避免此类任务在执行的时候对其他的前台任务的性能产生影响。TiKV 通过实时地监测所有前台任务所消耗的 CPU 和 IO 等资源，并根据实例总的资源上限计算出后台任务可使用的资源阈值，所有后台任务在执行时会受此阈值的限制。
```

**Review by Connor1996 — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-alter-resource-group.md [hunk 9]
```diff
@@ -57,8 +60,17 @@ ResourceGroupRunawayActionOption ::=
...
+BackgroundOptionList ::= 
+    DirectBackgroundOption
+|   BackgroundOptionList DirectBackgroundOption
+|   BackgroundOptionList ',' DirectBackgroundOption
+
+DirectBackgroundOption ::= 
+    "TASK_TYPES" EqOpt stringLit
 ```
 
+
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-resource-group.md:73]:
```suggestion
```

File: tidb-resource-control.md [hunk 10]
```diff
@@ -333,6 +333,53 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+
+自 v7.4.0 版本开始，TiDB 资源管控引入了对后台任务的管理，当一种任务被标记为 `后台任务` 时，TiKV 会对这种任务的执行施加动态的使用资源的限制，以尽量避免此类任务在执行的时候对其他的前台任务的性能产生影响。TiKV 通过实时地监测所有前台任务所消耗的 CPU 和 IO 等资源，并根据实例总的资源上限计算出后台任务可使用的资源阈值，所有后台任务在执行时会受此阈值的限制。
+
+#### `BACKGROUND` 参数说明
+
+`TASK_TYPES`：设置需要作为后台任务管理的任务类型，多个任务类型以 `,` 分隔。
+
+目前 TiDB 支持如下几种后台任务的类型：
+
+- `lightning`：使用 [TiDB Lightning](tidb-lightning/tidb-lightning-overview.md) 执行导入任务。同时支持 TiDB Lightning 的物理和逻辑导入模式。
```
Comment (glorv) [inline tidb-resource-control.md:352]:
```suggestion
- `lightning`：使用 [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md) 执行导入任务。同时支持 TiDB Lightning 的物理和逻辑导入模式。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: tidb-resource-control.md [hunk 11]
```diff
@@ -333,6 +333,53 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+目前 TiDB 支持如下几种后台任务的类型：
+
+- `lightning`：使用 [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md) 执行导入任务。同时支持 TiDB Lightning 的物理和逻辑导入模式。
+- `br`：使用 [BR](/br/backup-and-restore-overview.md) 执行数据备份和恢复。目前不支持 PITR。
+- `ddl`：对于 Reorg DDL，控制批量数据回写阶段的资源使用。
+- `analyze`：对应手动执行或系统自动触发的 [收集统计信息](/statistics.md#统计信息的收集) 任务。
+
+#### 示例
+
+1. 创建 `rg1` 资源组，并将后台任务类型设置为 `br` 和 `analyze`。
```
Comment (songrijie) [inline tidb-resource-control.md:359]:
```suggestion
1. 创建 `rg1` 资源组，并将 `br` 和 `analyze` 标记为后台任务。
```

File: tidb-resource-control.md [hunk 12]
```diff
@@ -333,6 +333,53 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+
+    ```sql
+    CREATE RESOURCE GROUP IF NOT EXISTS rg1 RU_PER_SEC = 500 BACKGROUND=(TASK_TYPES='br,analyze');
+    ```
+
+2. 修改 `rg1` 资源组，将后台任务类型修改为 `br` 和 `ddl`。
```
Comment (songrijie) [inline tidb-resource-control.md:365]:
```suggestion
2. 修改 `rg1` 资源组，将 `br` 和 `ddl` 标记为后台任务。
```

File: tidb-resource-control.md [hunk 13]
```diff
@@ -333,6 +333,53 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+
+`TASK_TYPES`：设置需要作为后台任务管理的任务类型，多个任务类型以 `,` 分隔。
+
+目前 TiDB 支持如下几种后台任务的类型：
+
+- `lightning`：使用 [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md) 执行导入任务。同时支持 TiDB Lightning 的物理和逻辑导入模式。
+- `br`：使用 [BR](/br/backup-and-restore-overview.md) 执行数据备份和恢复。目前不支持 PITR。
+- `ddl`：对于 Reorg DDL，控制批量数据回写阶段的资源使用。
+- `analyze`：对应手动执行或系统自动触发的 [收集统计信息](/statistics.md#统计信息的收集) 任务。
+
```
Comment (songrijie) [inline tidb-resource-control.md:356]: 需要补充说明认情况下的设置是什么。

**Review by nolouch — APPROVED** [review]
lgtm

File: tidb-resource-control.md (hunk 13 above)
Comment (glorv) [reply tidb-resource-control.md:358]: done

File: tidb-resource-control.md [hunk 14]
```diff
@@ -333,6 +333,55 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+#### `BACKGROUND` 参数说明
+
+`TASK_TYPES`：设置需要作为后台任务管理的任务类型，多个任务类型以 `,` 分隔。
+
+目前 TiDB 支持如下几种后台任务的类型：
+
+- `lightning`：使用 [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md) 执行导入任务。同时支持 TiDB Lightning 的物理和逻辑导入模式。
+- `br`：使用 [BR](/br/backup-and-restore-overview.md) 执行数据备份和恢复。目前不支持 PITR。
+- `ddl`：对于 Reorg DDL，控制批量数据回写阶段的资源使用。
+- `analyze`：对应手动执行或系统自动触发的[收集统计信息](/statistics.md#统计信息的收集) 任务。
```
Comment (Oreoxmt) [inline tidb-resource-control.md:355]:
```suggestion
- `analyze`：对应手动执行或系统自动触发的[收集统计信息](/statistics.md#统计信息的收集)任务。
```

File: tidb-resource-control.md [hunk 15]
```diff
@@ -333,6 +333,55 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+
+在默认情况下，被标记为后台任务的任务类型为空，此时可以任务后台任务的管理功能处于关闭状态，其行为和旧版本保持一致。用户需要手动修改 `default` 资源组的后台任务类型配置以开启后台任务的管理。
```
Comment (Oreoxmt) [inline tidb-resource-control.md:357]:
```suggestion
默认情况下，被标记为后台任务的任务类型为空，此时后台任务的管理功能处于关闭状态，其行为与 TiDB v7.3.0 之前版本保持一致。你需要手动修改 `default` 资源组的后台任务类型以开启后台任务管理。
```

**Review by Oreoxmt — APPROVED** [review]

File: tidb-resource-control.md [hunk 16]
```diff
@@ -333,6 +333,55 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+`TASK_TYPES`：设置需要作为后台任务管理的任务类型，多个任务类型以 `,` 分隔。
+
+目前 TiDB 支持如下几种后台任务的类型：
+
+- `lightning`：使用 [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md) 执行导入任务。同时支持 TiDB Lightning 的物理和逻辑导入模式。
+- `br`：使用 [BR](/br/backup-and-restore-overview.md) 执行数据备份和恢复。目前不支持 PITR。
+- `ddl`：对于 Reorg DDL，控制批量数据回写阶段的资源使用。
+- `analyze`：对应手动执行或系统自动触发的[收集统计信息](/statistics.md#统计信息的收集)任务。
+
+默认情况下，被标记为后台任务的任务类型为空，此时后台任务的管理功能处于关闭状态，其行为与 TiDB v7.3.0 之前版本保持一致。你需要手动修改 `default` 资源组的后台任务类型以开启后台任务管理。
```
Comment (Oreoxmt) [inline tidb-resource-control.md:357]:
```suggestion
默认情况下，被标记为后台任务的任务类型为空，此时后台任务的管理功能处于关闭状态，其行为与 TiDB v7.4.0 之前版本保持一致。你需要手动修改 `default` 资源组的后台任务类型以开启后台任务管理。
```

**Review by songrijie — COMMENTED** [review]
LGTM

File: tidb-resource-control.md [hunk 17]
```diff
@@ -333,6 +333,55 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
 
 + `information_schema.runaway_watches` 表中包含了 Runaway Queries 的快速识别规则记录。详见 [`RUNAWAY_WATCHES`](/information-schema/information-schema-runaway-watches.md)。
 
+### 管理后台任务
+
+> **警告：**
+>
+> 该功能目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```
Comment (ran-huang) [inline tidb-resource-control.md:340]: 这个链接是不是更好点
```suggestion
> 该功能目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues/new/choose) 反馈。
```

File: tidb-resource-control.md [hunk 18]
```diff
@@ -333,6 +333,55 @@ Runaway Query 是指执行时间或消耗资源超出预期的查询。下面使
...
+
+后台任务是指那些优先级不高但是需要消耗大量资源的任务，如数据备份和自动统计信息收集等。这些任务通常定期或不定期触发，在执行的时候会消耗大量资源，从而影响在线的高优先级任务的性能。
+
+自 v7.4.0 开始，TiDB 资源管控引入了对后台任务的管理。当一种任务被标记为后台任务时，TiKV 会对这种任务的执行施加动态的资源使用限制，以尽量避免此类任务在执行时对其他前台任务的性能产生影响。TiKV 通过实时地监测所有前台任务所消耗的 CPU 和 IO 等资源，并根据实例总的资源上限计算出后台任务可使用的资源阈值，所有后台任务在执行时会受此阈值的限制。
```
Comment (ran-huang) [inline tidb-resource-control.md:344]:
```suggestion
自 v7.4.0 开始，TiDB 资源管控引入了对后台任务的管理。当一种任务被标记为后台任务时，TiKV 会动态地限制该任务的资源使用，以尽量避免此类任务在执行时对其他前台任务的性能产生影响。TiKV 通过实时地监测所有前台任务所消耗的 CPU 和 IO 等资源，并根据实例总的资源上限计算出后台任务可使用的资源阈值，所有后台任务在执行时会受此阈值的限制。
```

**Review by ran-huang — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
@Oreoxmt @ran-huang  can we merge it?

**Pull request merged**